### PR TITLE
Add help command that displays list of commands

### PR DESCRIPTION
**Pull request by yotamN**
Together with some refactor to make it possible, I've added an help message (and an unload message that was declared but not implemented).

I've developed it over my other patch of auto java perform because it touched the same code area and I wanted to minimize merge conflicts later, hope it's okay.

### CODE REVIEW

File: frida_tools/_magics.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import abc
+import json
+
```
Comment (oleavr) [inline frida_tools/_magics.py:3]: Shouldn't there be two blank lines here according to PEP-8? (I might me misremembering but at least this is the convention I've used — so hope I didn't get it wrong 😅)

File: frida_tools/_magics.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        return 0
+
+    def execute(self, repl, args):
+        repl._reactor.schedule(lambda: self._resume())
+
+
+class Reload(Magic):
+    @property
+    def description(self):
+        return "reload (i.e. rerun) the script that was given as an argument to the repl"
```
Comment (oleavr) [inline frida_tools/_magics.py:33]:
```suggestion
        return "reload (i.e. rerun) the script that was given as an argument to the REPL"
```

File: frida_tools/_magics.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        pass
+
+    def execute(self, repl, args):
+        pass
+
+
+class Resume(Magic):
+    @property
+    def description(self):
+        return "sesume execution of the attached process"
```
Comment (oleavr) [inline frida_tools/_magics.py:20]:
```suggestion
        return "resume execution of the spawned process"
```

File: frida_tools/_magics.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        return 0
+
+    def execute(self, repl, args):
+        repl._unload_script()
+
+
+class Autoperform(Magic):
+    @property
+    def description(self):
+        return "receive on/off as first and only argument, when switched on will wrap any repl code with Java.PerformNow"
```
Comment (oleavr) [inline frida_tools/_magics.py:60]:
```suggestion
        return "receive on/off as first and only argument, when switched on will wrap any REPL code with Java.performNow()"
```

File: frida_tools/_magics.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                const _endTime = Date.now();
+                console.log('Time: ' + (_endTime - _startTime) + ' ms.');
+                return _result;
+            }})();'''.format(expression=json.dumps(" ".join(args))))
+
+
+class Help(Magic):
+    @property
+    def description(self):
+        return "print a list of available repl commands"
```
Comment (oleavr) [inline frida_tools/_magics.py:93]:
```suggestion
        return "print a list of available REPL commands"
```

File: frida_tools/_magics.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
+import abc
```
Comment (oleavr) [inline frida_tools/_magics.py:1]: Perhaps we should name this file `_repl_magic` or something? (Plural sounds a bit strange so perhaps singular is better?)

**Review by oleavr — CHANGES_REQUESTED** [review]
Wow, nice. Thanks!

**Comment by yotamN** [comment]
Fixed everything and rebased

**Comment by oleavr** [comment]
Thanks! Conflicts ⚔️

**Comment by yotamN** [comment]
Fixed

**Comment by oleavr** [comment]
Conflicts ⚔️

**Comment by yotamN** [comment]
rebased again

**Comment by starplanet** [comment]
There is a bug that _repl_magic.py is not listed in meson.build file. It will cause frida to fail to compile

**Pull request merged**